dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/metadata/DefaultMetadataExportService.java

```diff
@@ -208,8 +208,10 @@ public RootNode getMetadataAsNode( MetadataExportParams params )
 
         for ( Class<? extends IdentifiableObject> klass : metadata.keySet() )
         {
-            CollectionNode collectionNode = fieldFilterService.toCollectionNode( klass,
-                new FieldFilterParams( metadata.get( klass ), params.getFields( klass ), params.getDefaults(), params.getSkipSharing() ) );
+            FieldFilterParams fieldFilterParams = new FieldFilterParams( metadata.get( klass ), params.getFields( klass ), params.getDefaults(), params.getSkipSharing() );
+            fieldFilterParams.setUser( params.getUser() );
+
```

Comment:
This is a performance fix,  FieldFilterParams was missing user and  fieldFilterService.toCollectionNode() calling currentUserService in for loop